File: 哈希表/快乐数.go
package 哈希表

func isHappy(n int) bool {
	seen := map[int]bool{}
	for n != 1 && !seen[n] {
		seen[n] = true
		n = dwsfcewvew(n)
	}
	return n == 1
}

func dwsfcewvew(n int) int {
	sum := 0
	for n > 0 {
		sum += (n % 10) * (n % 10)
		n /= 10
	}
	return sum
}

File: 模拟/快乐数.java
package 模拟;

import java.util.HashSet;

public class 快乐数 {
    public boolean isHappy(int n) {
        var seen = new HashSet<>();
        seen.add(n);
        int num = 0;
        while(true){
             num = 0;
            while(n>0){
                num += (n%10)*(n%10);
                n/=10;
            }
            if (num == 1) return true;
            if (seen.contains(num)){
                return false;
            }else{
                seen.add(num);
            }
            n=num;
        }
    }
}
